src/plugins/audioplayers/plugin.c

```diff
@@ -217,4 +318,14 @@ static struct audio_player *audioplayers_linux_plugin_get_player(char *player_id
     return player;
 }
 
+static void audioplayers_linux_plugin_dispose_player(struct audio_player *player) {
+    list_for_each_entry_safe(struct audio_player_entry, entry, &plugin.players, entry) {
```

Comment:
@ardera Is this the only way to go about deleting particular instance of player?

Reply:
You could also add a `struct list_head` to the `audio_player`:
```C
struct audio_player {
    struct list_head entry;
    // ...
}
```

and make the `struct audio_player` declaration public (put it in a header file so that `plugin.c` can see it)

Then the removal from the list would just be as easy as `list_del(&audio_player->entry)` and the other list operations would be easier as well.

The only downside is that this isn't nicely encapsulated, i.e. in my mind the audio player itself shouldn't know what channels it is subscribed too. But that might be a bit pedantic, feel free to change it if you like :)

Reply:
Well, I do not mind current approach as it is very rare operation.
I will follow current style to encapsulate plugin as much as possible